types/react-router/test/RouterContext.tsx

```diff
@@ -51,7 +51,7 @@ const MultiStepSignup: React.FC = () => {
     }, [location.pathname]);
 
     const handleNextStep = React.useCallback(
-        event => {
+        (event: React.FormEvent<HTMLFormElement>) => {
```

Comment:
It cannot be any anymore

Reply:
That means to me we can't go ahead. I guess forking TS 2.x typings and moving latest to 3.x is the better option here. 

Though I'm not sure it's worth the breaking change (with migration part) just so that you get errors for `(untyped, typed: number) => void`. I've never seen this.

Reply:
I am not sure I understand.
The whole point of this PR is to get error if you pass a callback with implicity 'any' to useCallback

If the above code was not using `useCallback` but just declaring a normal function TS would complain 'event' has implicity type any.
```
const handleNextStep = event => {
// ....
}
```

Reply:
> Though I'm not sure it's worth the breaking change (with migration part) just so that you get errors for (untyped, typed: number) => void. I've never seen this.
Maybe my example was bad. We should get error also for `(untyped1, untyped2) => void` passed in useCallback

Reply:
@eps1lon To the best of my understanding the fix is valid and should be merged.

Allowing `implicit any` here is not a feature: it allows `any` to infect `no-implicit-any` codebases unnoticed when explicit typing is forgotten, which is never intended.
It happened countless times in my codebase and caused a few bugs.